### PR TITLE
feat(core): add `PATCH/GET /saml-applications/:id` APIs

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add following APIs:
1. PATCH /saml-applications/:id
2. GET /saml-applications/:id

This PR resolves LOG-10115 and LOG-10116

## Testing

Add integration tests.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      expiresAt: Math.floor(notAfter.getTime() / 1000),
+      active: false,
+    });
+  };
+
+  const findSamlApplicationById = async (id: string): Promise<SamlApplicationResponse> => {
+    const application = await findApplicationById(id);
+    assertThat(application.type === ApplicationType.SAML, 'application.saml.saml_application_only');
+
+    const samlConfig = await findSamlApplicationConfigByApplicationId(application.id);
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:47]: promise all?

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    return ensembleSamlApplication({ application, samlConfig });
+  };
+
+  const updateSamlApplicationById = async (
+    id: string,
+    patchApplicationObject: PatchSamlApplication
+  ): Promise<SamlApplicationResponse> => {
+    const { name, description, customData, config } = patchApplicationObject;
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:56]:
```suggestion
    const { config, ...applicationData } = patchApplicationObject;
```

File: packages/core/src/saml-applications/libraries/utils.ts [hunk 3]
```diff
@@ -56,3 +66,35 @@ const createCertificate = (keypair: forge.pki.KeyPair, lifeSpanInDays: number) =
...
+  return {
+    ...application,
+    ...samlConfig,
+  };
+};
+
+/**
+ * Only HTTP-POST binding is supported for receiving SAML assertions at the moment.
+ */
+export const validateAcsUrl = (acsUrl: SamlAcsUrl) => {
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/utils.ts:92]: Do we need to validate the URL format here?

File: packages/core/src/saml-applications/queries/configs.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ export const createSamlApplicationConfigQueries = (pool: CommonQueryMethods) =>
   const updateSamlApplicationConfig = buildUpdateWhereWithPool(pool)(SamlApplicationConfigs, true);
 
   const findSamlApplicationConfigByApplicationId = async (applicationId: string) =>
-    pool.maybeOne<SamlApplicationConfig>(sql`
+    pool.one<SamlApplicationConfig>(sql`
```
Comment (simeng-li) [inline packages/core/src/saml-applications/queries/configs.ts:19]: This config should be nullable?

File: packages/core/src/saml-applications/routes/index.ts [hunk 5]
```diff
@@ -11,7 +12,7 @@ import koaGuard from '#src/middleware/koa-guard.js';
 import { buildOidcClientMetadata } from '#src/oidc/utils.js';
 import { generateInternalSecret } from '#src/routes/applications/application-secret.js';
 import type { ManagementApiRouter, RouterInitArgs } from '#src/routes/types.js';
-import { ensembleSamlApplication, validateAcsUrl } from '#src/saml-applications/routes/utils.js';
+import { ensembleSamlApplication, validateAcsUrl } from '#src/saml-applications/libraries/utils.js';
```
Comment (simeng-li) [inline packages/core/src/saml-applications/routes/index.ts:15]: It's better to use relative paths within saml-application.
Additionally, consider moving these two utility methods out of the`/libraries/utils/` directory. `/libraries/utils/` is intended for use exclusively within the libraries. We could have a `/saml-applications/utils.js`

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      privateKey,
+      certificate,
+      expiresAt: Math.floor(notAfter.getTime() / 1000),
+      active: false,
+    });
+  };
+
+  const findSamlApplicationById = async (id: string): Promise<SamlApplicationResponse> => {
+    const application = await findApplicationById(id);
+    assertThat(application.type === ApplicationType.SAML, 'application.saml.saml_application_only');
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:45]: status: 422

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  };
+
+  const updateSamlApplicationById = async (
+    id: string,
+    patchApplicationObject: PatchSamlApplication
+  ): Promise<SamlApplicationResponse> => {
+    const { name, description, customData, config } = patchApplicationObject;
+    const originalApplication = await findApplicationById(id);
+    assertThat(
+      originalApplication.type === ApplicationType.SAML,
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:59]: ditto: status: 422

File: packages/core/src/saml-applications/queries/configs.ts (hunk 4 above)
Comment (darcyYe) [reply packages/core/src/saml-applications/queries/configs.ts:19]: When creating a SAML app, a record in the `saml_application_configs` table is inserted simultaneously. Subsequent operations only involve DB "update" actions on this record.
If this record is not inserted during the creation of the SAML app, additional checks will be required when updates to the config are needed later. This means that we can always expect this query to retrieve exact ONE record.

Comment (darcyYe) [reply packages/core/src/saml-applications/queries/configs.ts:19]: According to the design of the console, when creating a SAML app, you only need to provide `name`/`description`.

File: packages/core/src/saml-applications/routes/index.ts (hunk 5 above)
Comment (darcyYe) [reply packages/core/src/saml-applications/routes/index.ts]: Missed the last part. Will reorg the `utils.ts` file before merging.

File: packages/core/src/saml-applications/queries/configs.ts (hunk 4 above)
Comment (simeng-li) [reply packages/core/src/saml-applications/queries/configs.ts:19]: name/description should be stored in the application table?

**Review by simeng-li — APPROVED** [review]
overall LGTM

**Pull request merged**